### PR TITLE
Added more detailed documentation on Contributing

### PR DESCRIPTION
**Pull request by DirefulBelt6803**
I added documentation explaining how to contribute in detail. This documentation now lives on the pumpkin website as well as the DOCUMENTATION.md file

### CODE REVIEW

**Comment by Bryntet** [comment]
Pls rebase so there is not spam of "merge branch 'Snowiiii:master' into master"

File: docs/developer/contributing.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+- First create a github account if you don't already have one
+ 
+- Go to Pumpkin's Official [Github Repository](https://github.com/Snowiiii/Pumpkin) and press fork
+
+> Creating a fork means you now have your own copy of the Pumpkin source code, (this does not mean you own the copyright)
+
+  Now that you have a copy that you can edit, you will need a few tools
+
+- Install [Github-Desktop](https://desktop.github.com/download/) or [git](https://git-scm.com/downloads) for your operating system
```
Comment (Snowiiii) [inline docs/developer/contributing.md:34]: 1. Git is required to use GitHub desktop.
2. We should make it more clear that for newbies GitHub Desktop is an recommendation for easier use, not a must

File: docs/developer/contributing.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+> Github Desktop is a much easier tool to use. If you are unsure of which to use, choose this.
+
+- Install Rust at [rust-lang.org](https://www.rust-lang.org/).
+
+- Finally, if you are wanting to contribute to documentation, install [NodeJS](https://nodejs.org/en)
```
Comment (Snowiiii) [inline docs/developer/contributing.md:40]: I would split this up. If you only want to contribute to docs, you only need node. If you only want to change rust code you only need rust

File: docs/developer/contributing.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+- Use **git commit** to Commit your code to Master
+
+  On Github you can view your fork, if it says you are behind in contributions, click **Sync Fork** and **Update Branch**
+
+- Use **git pull** to update your local code
+
+  Ensure your code adheres to our project structure and style guidelines.
+
+  Write clear and concise commit messages that describe your changes.
```
Comment (Snowiiii) [inline docs/developer/contributing.md:76]: I think its is way too much, There are many explanations for how to use git/github desktop out there

**Pull request closed without merging**